slice/skip/skip.go

```diff
@@ -171,9 +260,44 @@ func (sl *SkipList) Insert(entries ...Entry) Entries {
 	return overwritten
 }
 
+func (sl *SkipList) insertAtPosition(position uint64, entry Entry) {
+	if position > sl.num {
+		position = sl.num
+	}
+	sl.cache.reset()
+	sl.posCache.reset()
+	n, pos := sl.searchByPosition(position, sl.cache, sl.posCache)
+	insertNode(sl, n, entry, pos, sl.cache, sl.posCache, true)
+}
+
+// InsertAtPosition will insert the provided entry and the provided position.
+// If position is greater than the length of the skiplist, the entry
+// is appended.  This method bypasses order checks and checks for
+// duplicates so use with caution.
+func (sl *SkipList) InsertAtPosition(position uint64, entry Entry) {
+	sl.insertAtPosition(position, entry)
+}
```

Comment:
From what I can see, this clears the cache on every call. Will inserting 100 items cause 100 cache clears and rebuilds? If so, would it be worth it to make a variadic Insertion call?